main/realmain.cc

```diff
@@ -201,11 +202,15 @@ void runAutogen(const core::GlobalState &gs, options::Options &opts, const autog
 
     auto resultq = make_shared<BlockingBoundedQueue<AutogenResult>>(indexed.size());
     auto fileq = make_shared<ConcurrentBoundedQueue<int>>(indexed.size());
+    // all we're doing here is adding to a vector, rather than consuming packages as they're produced
```

Comment:
This mutex seems a bit like overkill.

I think you can simplify this code with a simple serial pass thru the files to identify and parse packages. Non-package files can be pushed to `fileq`. If this becomes a serial bottleneck (once we have many packages), we can have a parallel package parsing phase followed by a parallel `runAutogen` phase (still not requiring a mutex; we can use a concurrent queue).